File: v3/typescript/api/search-similar-api.ts
import { SearchSimilarApiGenerated, SearchSimilarApiGetRequest, SearchSimilarApiPostRequest } from "./search-similar-api-generated";
import { searchAll } from "../search-all-helper";
import { AxiosRequestConfig } from "axios";
export * from "./search-similar-api-generated";

export class SearchSimilarApi extends SearchSimilarApiGenerated {

    public async get_all(requestParameters: SearchSimilarApiGetRequest, options?: AxiosRequestConfig) {
        return searchAll(this.get.bind(this), requestParameters, options);
    }

    public async post_all(requestParameters: SearchSimilarApiPostRequest, options?: AxiosRequestConfig) {
        return searchAll(this.post.bind(this), requestParameters, options);
    }
}

File: v3/python/newscatcherapi_client/apis/tags/search_similar_api.py
# coding: utf-8

from newscatcherapi_client.apis.tags.search_similar_api_generated import SearchSimilarApiGenerated
from newscatcherapi_client.search_all_helper import search_all

class SearchSimilarApi(SearchSimilarApiGenerated):

    def get_all(self, **kwargs):
        return search_all(self.get, **kwargs)

    def post_all(self, **kwargs):
        return search_all(self.post, **kwargs)
